code/game/machinery/machinery.dm

```diff
@@ -130,6 +130,9 @@ Class Procs:
 	var/has_special_power_checks = FALSE	// If true, call auto_use_power instead of doing it all in SSmachinery.
 	var/clicksound //played sound on usage
 	var/clickvol = 40 //volume
+	var/listener/button_sig_listener
+	var/id = null
+	var/button_control = FALSE
```

Comment:
I'm not sold on this variable. Two reasons. First, in order to make something listener controllable you now have to modify TWO variables. ID AND this. Is there ever a case where you'd make ID be not-null but leave this as FALSE?

Reply:
yes, ID is a variable used by lots of machinery interally. Like T-comms, scrubbers, etc. So I just needed a different variable to be introduced to indicate whether we use ID for remote control or not. Probably a better way may be to make another war like `button_ID` remove ID usage